analyzer/src/main/kotlin/managers/MavenDependencyHandler.kt

```diff
@@ -0,0 +1,122 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.analyzer.managers
+
+import org.apache.maven.project.MavenProject
+import org.apache.maven.project.ProjectBuildingException
+
+import org.eclipse.aether.graph.DependencyNode
+
+import org.ossreviewtoolkit.analyzer.managers.utils.DependencyHandler
+import org.ossreviewtoolkit.analyzer.managers.utils.MavenSupport
+import org.ossreviewtoolkit.analyzer.managers.utils.identifier
+import org.ossreviewtoolkit.model.OrtIssue
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.PackageLinkage
+import org.ossreviewtoolkit.model.createAndLogIssue
+import org.ossreviewtoolkit.utils.collectMessagesAsString
+import org.ossreviewtoolkit.utils.log
+import org.ossreviewtoolkit.utils.showStackTrace
+
+/**
+ * A specialized [DependencyHandler] implementation for the dependency model of Maven.
+ */
+class MavenDependencyHandler(
+    /** The name of the associated package manager. */
+    val managerName: String,
+
+    /** The helper object to invoke Maven-related functionality. */
+    val mvn: MavenSupport,
+
+    /**
+     * A map with information about the local projects in the current Maven build. Dependencies pointing to projects
+     * sometimes need to be treated in a special way.
+     */
+    val localProjects: Map<String, MavenProject>,
+
+    /**
+     * A flag whether [SBT compatibility mode][Maven.enableSbtMode] is enabled.
+     */
+    val sbtMode: Boolean
+) : DependencyHandler<DependencyNode> {
+    override fun identifierFor(dependency: DependencyNode): String {
+        val id = dependency.identifier()
+        val typePrefix = if (isLocalProject(id)) managerName else "Maven"
+        return "$typePrefix:$id"
+    }
+
+    override fun dependenciesFor(dependency: DependencyNode): Collection<DependencyNode> {
+        val children = dependency.children
+        val filteredChildren = children.filterNot { node ->
+            TOOL_DEPENDENCIES.any(node.artifact.identifier()::startsWith)
+        }
+
+        if (filteredChildren.size < children.size) {
+            log.info { "Omitting the Java < 1.9 system dependency on 'tools.jar'." }
+        }
+
+        return filteredChildren
+    }
+
+    override fun linkageFor(dependency: DependencyNode): PackageLinkage =
+        if (isLocalProject(dependency)) PackageLinkage.PROJECT_DYNAMIC
+        else PackageLinkage.DYNAMIC
+
+    override fun createPackage(
+        identifier: String,
```

Comment:
Not directly related to this commit, but what was the reason to use `String` instead of `Identifier` in the `DependencyHandler`?

Reply:
To be honest, I am not entirely sure any more. IIRC, in the original implementation there was no direct 1:1 match between this identifier string and an `Identifier`, but this is now indeed the case. There is at least the advantage that the serialization of a plain string takes less space than an `Identifier` object.

Reply:
So, are you planning to change this in this commit?

Reply:
As this change would affect other functions of the `DependencyHandler` interface and the `DependencyGraphBuilder` class as well, I would rather think some more about it and do it in a separate PR - if at all.

Reply:
Somewhat related: https://github.com/oss-review-toolkit/ort/pull/4030